registry/storage/manifeststore.go

```diff
@@ -124,6 +126,16 @@ func (ms *manifestStore) verifyManifest(ctx context.Context, mnfst *manifest.Sig
 	}
 
 	if !ms.skipDependencyVerification {
+		if len(mnfst.FSLayers) == 0 || len(mnfst.History) == 0 {
+			errs = append(errs, distribution.ErrManifestValidation{
+				Reason: "no layers present"})
+		}
+
+		if len(mnfst.FSLayers) != len(mnfst.History) {
```

Comment:
@endophage Per the bug, I believe history and fslayers will always have the same link. We detect the condition of partial manifests through link traversal, below.

However, there are cases when the parent isn't set. Really we can only detect if the parent is set and it is not adjacent in the history entries.